test/e2e/framework/network/utils.go

```diff
@@ -334,6 +333,36 @@ func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containe
 	return eps, nil
 }
 
+// GetResponseFromContainer executes a curl via kubectl exec in a container.
+// If GetResponseFromContainer returns nil, an error will be logged by framework.
+func (config *NetworkingTestConfig) GetResponseFromContainer(protocol, dialCommand, containerIP, targetIP string, containerHTTPPort, targetPort int) *NetexecDialResponse {
+	ipPort := net.JoinHostPort(containerIP, strconv.Itoa(containerHTTPPort))
+	cmd := makeCURLDialCommand(ipPort, dialCommand, protocol, targetIP, targetPort)
+
+	stdout, stderr, err := config.f.ExecShellInPodWithFullOutput(config.TestContainerPod.Name, cmd)
+	if err != nil {
+		// A failure to kubectl exec counts as a try, not a hard fail.
+		// Also note that we will keep failing for maxTries in tests where
+		// we confirm unreachability.
+		framework.Logf("Failed to execute %q: %v, stdout: %q, stderr: %q", cmd, err, stdout, stderr)
+		return nil
```

Comment:
Returning nil on error is misleading.

Instead of that - I would really prefer returning (*NetexecDialResponse, error) from this method and returning error explicitly here.